disnake/shard.py

```diff
@@ -4,7 +4,19 @@
 
 import asyncio
 import logging
-from typing import TYPE_CHECKING, Any, Callable, Dict, List, Optional, Tuple, Type, Union, overload
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    Dict,
+    List,
+    NoReturn,
```

Comment:
```suggestion
    NoReturn as Never,
```
makes the intent clear

Reply:
I think we should add this to disnake/utils.py or something, aliased there.

Reply:
A separate file would be good, maybe `disnake/_types.py` to match what ext.commands already does.

Reply:
Actually, this is fine in the same file

Reply:
https://github.com/DisnakeDev/disnake/pull/782/commits/3e509f78f7caeaa3e011542b944d3172f20605d8